kernel/src/engine/default/parquet.rs

```diff
@@ -124,6 +133,17 @@ impl FileOpener for ParquetOpener {
             let options = ArrowReaderOptions::new(); //.with_page_index(enable_page_index);
             let mut builder =
                 ParquetRecordBatchStreamBuilder::new_with_options(reader, options).await?;
+            if let Some(predicate) = predicate {
+                let parquet_schema = metadata.schema();
+                let parquet_physical_schema = metadata.parquet_schema();
+                let row_filter = expression_to_row_filter(
+                    predicate,
+                    &table_schema,
+                    parquet_schema,
+                    parquet_physical_schema,
+                )?;
+                builder = builder.with_row_filter(row_filter);
```

Comment:
I'm not sure this will work as well as we wish it would... 

In my experience, this kind of row-level pushdown doesn't consistently help performance -- even with the kind of lazy materialization arrow-rust brags about (it won't fetch column chunks until proving at least one row is needed).

The reason is: Every row pays the cost of evaluating the filter, while any I/O reduction is only partial at best. We still have to fetch the columns the predicate touches, so any I/O savings come by not fetching payload columns. But that only works if the filter eliminated _ALL_ rows from the row group. And if row groups can be skipped, then stats-based row group skipping can often do it much more cheaply (no extra I/O at all).

Meanwhile, in cases that don't see any I/O reduction, pushing down the filtering just shifts complexity from the query engine to the file scanner. And that's usually a net loss because the output of the scan is likely consumed in a pipelined single pass either way.

Reply:
Note: There are _absolutely_ cases where row-level filter pushdown is a performance win... but there are too many cases where it doesn't help or even hurts performance instead. And it's data-dependent, so hard to predict how any one query will be affected.

Reply:
thanks ryan planning on pausing this work for now :)